hail/python/hail/matrixtable.py

```diff
@@ -2953,13 +2958,12 @@ def _select_entries(self, caller, s) -> 'MatrixTable':
         return cleanup(MatrixTable(base._jvds.selectEntries(str(s._ir))))
 
     @typecheck_method(caller=str,
-                      row=expr_struct(),
-                      new_key=nullable(sequenceof(str)))
-    def _select_rows(self, caller, row, new_key=None):
+                      row=expr_struct())
+    def _select_rows(self, caller, row):
         analyze(caller, row, self._row_indices, {self._col_axis})
         base, cleanup = self._process_joins(row)
 
-        return cleanup(MatrixTable(base._jvds.selectRows(str(row._ir), new_key)))
+        return cleanup(MatrixTable(base._jvds.selectRows(str(row._ir))))
 
     @typecheck_method(key_struct=expr_struct())
     def _select_rows_processed(self, key_struct):
```

Comment:
I find all of these methods confusing how some take the key still, but others do not like `_select_rows`. Is there a way to unify / consolidate these or is this the minimum that's needed.

Reply:
Patrick just cleaned up the Table python internal interfaces (_select_scala), I think the same can be done with the matrix stuff

Reply:
I find it confusing too. I tried to simplify where I could, but it would take some time for me to get comfortable enough with the python side of things to make a comprehensive change. For example, it's really not clear to me what `_select_rows_processed` is for.